model/src/main/java/org/cloudfoundry/identity/uaa/zone/UserConfig.java

```diff
@@ -22,6 +22,8 @@ public class UserConfig {
 
     private List<String> defaultGroups = DEFAULT_ZONE_GROUPS;
 
+    private long maxUsers = -1;
```

Comment:
Why is `maxUsers` `long` instead of `int`? Do we need it to be more than max int value?

Reply:
because we can have more than 2 mio users and simply the compare functions should always run with long , simply to omit casts
A select count with int as return type means you never expect ot have more values in your DB but we expect to have many values :-)